src/http-gateways/trustless-gateway.md

```diff
@@ -66,12 +74,137 @@ Below response types MUST to be supported:
 - [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car) – disables IPLD/IPFS deserialization, requests a verifiable CAR stream to be returned
 - [application/vnd.ipfs.ipns-record](https://www.iana.org/assignments/media-types/application/vnd.ipfs.ipns-record) – requests a verifiable :cite[ipns-record] (multicodec `0x0300`).
 
+## Request Query Parameters
+
+### :dfn[dag-scope] (request query parameter)
+
+Optional, `dag-scope=(block|entity|all)` with default value `all`, only available for CAR requests.
+
+Describes the shape of the DAG fetched the terminus of the specified path whose blocks
+are included in the returned CAR file after the blocks required to traverse
+path segments.
+
+- `block` - Only the root block at the end of the path is returned after blocks
+  required to verify the specified path segments.
+
+- `entity` - For queries that traverse UnixFS data, `entity` roughly means return
+  blocks needed to verify the terminating element of the requested content path.
+  For UnixFS, all the blocks needed to read an entire UnixFS file, or enumerate a UnixFS directory.
+  For all queries that reference non-UnixFS data, `entity` is equivalent to `block`
+
+- `all` - Transmit the entire contiguous DAG that begins at the end of the path
+  query, after blocks required to verify path segments
+
+When present, returned `Etag` must include unique prefix based on the passed scope type.
+
+### :dfn[entity-bytes] (request query parameter)
+
+Optional, `entity-bytes=from:to` with the default value `0:*`, only available for CAR requests.
+Serves as a trustless form of an HTTP Range Request.
+
+When the terminating entity at the end of the specified content path can be
+interpreted as a continuous array of bytes (such as a UnixFS file), returns
+only the minimal set of blocks required to verify the specified byte range of
+said entity.
+
+Allowed values for `from` and `to` are positive integers where `to` >= `from`, which
+limit the return blocks to needed to satisfy the range `[from,to]`:
+
+- `from` value gives the byte-offset of the first byte in a range.
+- `to` value gives the byte-offset of the last byte in the range; that is,
+the byte positions specified are inclusive.  Byte offsets start at zero.
```

Comment:
What's the behaviour for a `from` or a `to` that takes us outside of the bounds of the file (positive `from` or `to` that is beyond EOF and negative `to` that resolves to <0?

Reply:
What is the behaviour if `from` and `to resolve to a zero byte range? does that mean you get no blocks, or just the block that the byte that they both point to?

Reply:
Thank you for pointing these gaps around error handling.

The rule of thumb we try to follow on trustless gateways is empowering the client to be able to self-verify the response, and that includes error scenarios. If a request was done for an invalid range, the response should include enough blocks for the client to see that.

@rvagg  I've added clarification in  608e7b7  – lmk if it is good enough  or how we could make it better.

(Not feeling strongly about zero-range requests, it is an edge case with no impact on real world requests, I made it collapse to be the same as `dag-scope=block` for the sake of simplicity, but lmk if other approach will be easier.)